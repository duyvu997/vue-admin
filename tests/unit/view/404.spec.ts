import { mount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import ElementUI from "element-ui";
import NotFound from "@/views/404.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(ElementUI);

const routes = [
  {
    path: "/404",
  }
];

const router = new VueRouter({
  routes
});

const wrapper = mount(NotFound, {
  localVue,
  router
});

describe("NotFound.vue", () => {
  it("snapshot testing", async () => {
    router.push("/404");
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot() 
 });
});
