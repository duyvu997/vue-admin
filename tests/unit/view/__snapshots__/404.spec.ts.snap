// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.vue snapshot testing 1`] = `
<div
  class="wscn-http404-container"
>
  <div
    class="wscn-http404"
  >
    <div
      class="pic-404"
    >
      <img
        alt="404"
        class="pic-404__parent"
        src="@/assets/404-images/404.png"
      />
       
      <img
        alt="404"
        class="pic-404__child left"
        src="@/assets/404-images/404-cloud.png"
      />
       
      <img
        alt="404"
        class="pic-404__child mid"
        src="@/assets/404-images/404-cloud.png"
      />
       
      <img
        alt="404"
        class="pic-404__child right"
        src="@/assets/404-images/404-cloud.png"
      />
    </div>
     
    <div
      class="text-404"
    >
      <div
        class="text-404__oops"
      >
        
        OOPS!
      
      </div>
       
      <div
        class="text-404__info"
      >
        
        All rights reserved
        
        <a
          href="https://wallstreetcn.com"
          style="color: rgb(32, 160, 255);"
          target="_blank"
        >
          wallstreetcn
        </a>
      </div>
       
      <div
        class="text-404__headline"
      >
        
        404 Page Not Found
      
      </div>
       
      <div
        class="text-404__info"
      >
        
        Please check that the URL you entered is correct, or click the button
        below to return to the homepage.
      
      </div>
       
      <a
        class="text-404__return-home"
        href=""
      >
        Back to home
      </a>
    </div>
  </div>
</div>
`;
